subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskExecution.java

```diff
@@ -423,9 +422,6 @@ public InputChangeTrackingStrategy getInputChangeTrackingStrategy() {
             if (taskAction instanceof IncrementalInputsTaskAction) {
                 return InputChangeTrackingStrategy.INCREMENTAL_PARAMETERS;
             }
-            if (taskAction instanceof IncrementalTaskInputsTaskAction) {
```

Comment:
We should also merge `AbstractIncrementalTaskAction` with `IncrementalInputsTaskAction`.

Reply:
Merged. Also merged `AbstractIncrementalTaskActionFactory` with `IncrementalInputsTaskActionFactory`